src/Symfony/Component/Form/FormInterface.php

```diff
@@ -303,7 +301,7 @@ public function handleRequest($request = null);
      *
      * @throws Exception\AlreadySubmittedException if the form has already been submitted
      */
-    public function submit($submittedData, bool $clearMissing = true);
+    public function submit(string|array|null $submittedData, bool $clearMissing = true);
```

Comment:
I think we should deprecate not passing anything else before doing this, or we might break some code, I'm even surprised  that tests are passing.

Reply:
This is the interface: implementations can still accept more types. See eg `Form`, where I put `mixed` here.